### PR TITLE
The chart component dependencies

### PR DESCRIPTION
**Pull request by lmacko**
https://github.com/Lukindo/richfaces-sandbox/tree/master/charts

### CODE REVIEW

File: examples/charts.html [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+    <script type="text/javascript" src="jquery-ui/ui/jquery.ui.menu.js"></script>
+
+    <script type="text/javascript" src="flot/jquery.flot.js"></script>
+    <script type="text/javascript" src="flot/jquery.flot.selection.js"></script>
+    <script type="text/javascript" src="flot/jquery.flot.pie.js"></script>
+    <script type="text/javascript" src="flot/jquery.flot.categories.js"></script>
+    <script type="text/javascript" src="flotTooltip/js/jquery.flot.tooltip.js"></script>
+
+    <script type="text/javascript" src="flotOrderBars/js/jquery.flot.orderBars.js"></script><!-- Must be before stack plugin-->
+    <script type="text/javascript" src="widgets/charts/charts.js"></script>
```
Comment (bleathem) [inline examples/charts.html]: This should point to widgets/output/chart.js ?

File: bower.json [hunk 2]
```diff
@@ -4,9 +4,13 @@
   "dependencies": {
     "jquery": "~1.9.0",
     "jquery-ui": "~1.9.2",
-    "bootstrap": "https://github.com/twitter/bootstrap/archive/3.0.0-wip.zip",
+    "bootstrap": "https://github.com/twbs/bootstrap/archive/v3.0.0-rc1.zip",
     "font-awesome": "~3.2.1",
     "modernizr": "~2.6.2",
-    "lodash": "~1.3.1"
+    "lodash": "~1.3.1",
+    "flot" : "0.8.0",
```
Comment (bleathem) [inline bower.json:11]: Can you rebase against master?  The bower.json file has changed since your fork.

File: examples/charts.html [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+        <div id="mulitplebar" class="rich-chart">
+        </div>
+
+        <h2>Multiple category bar chart</h2>
+        <div id="mulitplecategorybar" class="rich-chart">
+        </div>
+
+
+        <h2>Bar DateTime chart</h2>
+        <div id="datebar" class="rich-chart">
```
Comment (bleathem) [inline examples/charts.html]: The datebar and dateline examples are not implemented.  Is this an oversight?  If not, we should we exclude them/comment them out from the example, and create an richwidgets github issue to track their implementation.

**Comment by bleathem** [comment]
This look really good, nice work!  A few inline comments above to be resolved, and I think we are good to merge.  Moving forward we will probably want to review/revise the public widget API, as it currently seems limited in terms of what methods are available, but a great start!

File: bower.json (hunk 2 above)
Comment (bleathem) [inline bower.json:11]: Also, we are depending on a number of somwheat obscure github projects/forks here.  Any idea if there are issues tracking the inclusion of this functionality in the upstream flot project?  If not, we should ping the respective authors asking them if they intend to do so, and take up the torch ourselves if not.  Probably good to track this in a richwidgets github issue.

File: examples/charts.html (hunk 1 above)
Comment (lmacko) [inline examples/charts.html]: fixed.

File: bower.json (hunk 2 above)
Comment (lmacko) [inline bower.json:11]: I've done rebase. However, I was not able to run project after it there occurred an error during [grunt bower](http://pastebin.com/dyJQWYfe) and [grunt](http://pastebin.com/nsLQrWMM). I removed devdependencies from [bower.json](https://github.com/Lukindo/richfaces-widgets/commit/1ccbba140dfc28e7269b4ec1780d6f3f27bc7f75). Should I change something to be able to build with devDependencies?

Comment (lmacko) [inline bower.json:11]: As far as I know, the flot doesn't plan to include the plugins into the project. There is a subpage called [plugin repository](http://www.flotcharts.org/plugins/) but it is just a web page with links to developers repositories.

**Comment by bleathem** [comment]
Nice work with the Chart widgets @Lukindo !

**Pull request merged**